server/routes/visit.ts

```diff
@@ -71,35 +71,36 @@ export default function routes({
     const reference = getVisitReference(req)
     const fromPage = typeof req.query?.from === 'string' ? req.query.from : null
     const fromPageQuery = typeof req.query?.query === 'string' ? req.query.query : null
+    const { username } = res.locals.user
 
     const { visitHistoryDetails, visitors, notifications, additionalSupport } = await visitService.getFullVisitDetails({
       reference,
-      username: res.locals.user.username,
+      username,
     })
     const { visit } = visitHistoryDetails
     const filteredVisitHistoryDetails = visitHistoryDetails.eventsAudit.filter(event =>
       Object.keys(eventAuditTypes).includes(event.type),
     )
 
     if (visit.prisonId !== req.session.selectedEstablishment.prisonId) {
-      const visitPrison = await supportedPrisonsService.getPrison(res.locals.user.username, visit.prisonId)
+      const visitPrison = await supportedPrisonsService.getPrison(username, visit.prisonId)
 
       return res.render('pages/visit/summary', {
         visit: { reference: visit.reference },
         visitPrisonName: visitPrison.prisonName,
       })
     }
 
-    const prisoner: Prisoner = await prisonerSearchService.getPrisonerById(visit.prisonerId, res.locals.user.username)
-    const supportedPrisonIds = await supportedPrisonsService.getSupportedPrisonIds(res.locals.user.username)
+    const prisoner: Prisoner = await prisonerSearchService.getPrisonerById(visit.prisonerId, username)
+    const supportedPrisonIds = await supportedPrisonsService.getSupportedPrisonIds(username)
```

Comment:
Over-`await`ing here: could be a `Promise.all()` (appreciate this is existing code; only spotted it because of the `username` change...)